lang/src/org/partiql/lang/eval/ExprValueExtensions.kt

```diff
@@ -287,9 +293,25 @@ fun ExprValue.cast(
                         castFailedErr("can't convert string value to TIMESTAMP", internal = false, cause = e)
                     }
                 }
+                DATE -> when {
+                    type == TIMESTAMP -> {
+                        val ts = timestampValue()
+                        return valueFactory.newDate(LocalDate.of(ts.year, ts.month, ts.day))
+                    }
+                    type.isText -> try {
+                        val date = LocalDate.parse(stringValue())
```

Comment:
Strong +1 here--the casting code should be refactored to leverage the same parsing machinery as the compiler if at all possible.

Reply:
Good point. Changed. Included tests to reflect the change. Although I have a thought here. Since valid strings can be cast to `timestamp`, should we allow such strings to be cast to `date`?
For example, `CAST('2016-03-01T01:12:12Z' AS TIMESTAMP)`  returns timestamp.
Shouldn't `CAST('2016-03-01T01:12:12Z' AS DATE)` return `DATE '2016-03-01'`.
Currently, it doesn't. But we are supporting cast from timestamp to date.